### PR TITLE
Okta connector

### PR DESCRIPTION
**Pull request by oded-weber**
Required items, please complete
   
   Change(s):
   - Creating new Okta clv2 CCP connector, backward compatibility parser and ASIM parsers

   Reason for Change(s):
   - Adjusting Okta to clv2 and new CCP API

   Testing Completed:
   - I tested the connector but still need to test the parsers in canary
   - Also we need to update analytic rules and hunting query to work with the new parser

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Updated syntax for XYZ.yaml

   Reason for Change(s):
   - New schema used for XYZ.yaml
   - Resolves ISSUE #1234

   Version updated:
   - Yes
   - Detections/Analytic Rule templates are required to have the version updated

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Thanks for raising this PR. This PR will be investigated and will update you about the same before 13 Spetember,2023.

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Please repackage the solution using V3 tool.
https://github.com/Azure/Azure-Sentinel/blob/master/Tools/Create-Azure-Sentinel-Solution/V3/README.md

Also please share sample data to test the content of the solution.

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Please go through the changes requested above.

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Please go through the changes requested above.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @oded-weber, I've resolved the issue which you were facing while repackaging the solution. I committed the changes as well. Please review these commits.

**Comment by oded-weber** [comment]
Hi,
The changes look ok.
Is there something I should do in order to proceed?
Thanks,
Oded

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by v-prasadboke** [comment]
Yes, Please resolve the branch conflicts.

**Comment by v-prasadboke** [comment]
Hello @oded-weber, I was testing the solution but Sample data is missing. Can you please share sample data to test the parser also please resolve the Branch conflicts.
Thanks.

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Please provide requested content.

**Comment by v-prasadboke** [comment]
Hello @oded-weber, we are waiting for you

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Please share the sample data.

**Comment by oded-weber** [comment]
Please see attached.
Thanks,
Oded

**Comment by v-prasadboke** [comment]
Sorry @oded-weber but I dont see any files attached to the mail.

**Comment by oded-weber** [comment]
Here they are again.

**Comment by oded-weber** [comment]
[Okta Real Logs 1.txt](https://github.com/Azure/Azure-Sentinel/files/12847266/Okta.Real.Logs.1.txt)

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Thank you for sharing the sample data. Will test the solution and will update you about the same before 13 October, 2023.

**Comment by v-prasadboke** [comment]
Hello @oded-weber please update the branch from master, I'm KQL validation check is failing and it is not accessible by me.

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Please update the branch from master.

**Comment by oded-weber** [comment]
Done.

**Comment by v-prasadboke** [comment]
Thanks @oded-weber.

**Comment by v-prasadboke** [comment]
Hello @oded-weber sorry for the inconvenience but we are still working on this and will update you about the same before 24 October, 2023.

**Comment by oded-weber** [comment]
OK. Thanks!

File: Solutions/Okta Single Sign-On/Data Connectors/OktaNativePollerConnectorV2/azuredeploy_Okta_native_poller_connector_v2.json
[the remark this answers is not in the document]
Comment (oded-weber) [reply Solutions/Okta Single Sign-On/Data Connectors/OktaNativePollerConnectorV2/azuredeploy_Okta_native_poller_connector_v2.json:1]: 1) I do not understand. What is the earlier one? If it is the Azure function connector, then yes. We would like to have both.
2) What validation fails?

**Comment by v-prasadboke** [comment]
- Validation checks which are failing 
image

- About the custom connector you added a new one with V2. 
  Already the solution have 1 custom connector **OktaNativePollerConnector**  you added a new one as 
  **OktaNativePollerConnectorV2**.
  My concern here is, is the newly added custom connector is a modified version of the older one or a different one. 

- As this connector is a CCP connector which is not supported by V3 tool. We are working on this and will update you about 
  the same before 30 October, 2023.

**Comment by oded-weber** [comment]
As far as I understood, the old connector has never been published and is not relevant anymore ***@***.******@***.***> - please confirm).
Thanks,
Oded

**Comment by oded-weber** [comment]
Which validation checks are failing and why?

**Comment by v-prasadboke** [comment]
1.
```
Incorrect yaml file. File path: Parsers/ASimAuthentication/Parsers/ASimAuthenticationOktaOSS.yaml. Error message: can not read a block mapping entry; a multiline key may not be an implicit key at line 114, column 1:
    
    ^
Incorrect yaml file. File path: Parsers/ASimAuthentication/Parsers/vimAuthenticationOktaOSS.yaml. Error message: end of the stream or a document separator is expected at line 49, column 2:
    	, actor_alternateId_s = column_i ... 
     ^
```
2.
````
  Error Message:
   YamlDotNet.Core.SyntaxErrorException : (Line: 23, Col: 14, Idx: 671) - (Line: 26, Col: 1, Idx: 772): While scanning a block scalar, find a tab character where an indentation space is expected.
````

Error mentioned in number 2 is failing in all the files.

**Comment by oded-weber** [comment]
Hi, I do not understand what fails and I do not know how to fix it...
Did you try to run the parsers and they did not work?
Maybe there is a problem in the validation tests?
Please advise how we can proceed and who can help understanding the errors

**Comment by v-prasadboke** [comment]
Hello @oded-weber, Sorry for the late reply. I'll try to resolve this validation errors from our side.

**Comment by oded-weber** [comment]
Great. Thanks!

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by oded-weber** [comment]
Hi,
Any progress?
Thanks,
Oded

**Comment by oded-weber** [comment]
OK. Thanks!

**Comment by v-prasadboke** [comment]
Hello @oded-weber we are still working on packaging tool to get CCP connector in the maintemplate. Will provide you some update before 08 November, 2023. 
Thanks.

**Comment by oded-weber** [comment]
***@***.*** ***@***.***> & @Anki ***@***.***>.
Hi Prateek & Anki,
I need you help in accelerating the below.
We wanted to publish it as part of the October private previews. Now we are one month late and it is still WIP.
As far as I understand the connector is working so it is only a matter of packaging and publishing it. As we agreed the packaging should be done manually as we do not want to wait to the implementation of CCP in the packaging tool.
Thanks,
Oded

**Comment by v-prasadboke** [comment]
Hi @anki-narravula, Thanks for taking a look into this PR.

**Pull request merged**